tests/test_compression.py

```diff
@@ -119,5 +120,43 @@ def test_lzma_round_trip_format_raw(self, payload, filters):
 
 
 class TestZlib(unittest.TestCase):
-    # TODO: https://docs.python.org/3/library/zlib.html
-    pass
+    @given(payload=st.binary(), value=st.just(1) | st.integers())
+    def test_adler32(self, payload, value):
+        checksum = zlib.adler32(payload, value)
+        self.assertIsInstance(checksum, int)
+
+    @given(
+        payload_piece_1=st.binary(),
+        payload_piece_2=st.binary(),
+        value=st.just(1) | st.integers(),
+    )
+    def test_adler32_two_part(self, payload_piece_1, payload_piece_2, value):
+        combined_checksum = zlib.adler32(payload_piece_1 + payload_piece_2, value)
+        checksum_part1 = zlib.adler32(payload_piece_1, value)
+        checksum = zlib.adler32(payload_piece_2, checksum_part1)
+        self.assertEqual(combined_checksum, checksum)
+
+    @given(payload=st.binary(), value=st.just(0) | st.integers())
+    def test_crc32(self, payload, value):
+        crc = zlib.crc32(payload, value)
+        self.assertIsInstance(crc, int)
+
+    @given(
+        payload_piece_1=st.binary(),
+        payload_piece_2=st.binary(),
+        value=st.just(0) | st.integers(),
+    )
+    def test_crc32_two_part(self, payload_piece_1, payload_piece_2, value):
+        combined_crc = zlib.crc32(payload_piece_1 + payload_piece_2, value)
+        crc_part1 = zlib.crc32(payload_piece_1, value)
+        crc = zlib.crc32(payload_piece_2, crc_part1)
+        self.assertEqual(combined_crc, crc)
+
+    @given(
+        payload=st.binary(),
+        level=st.just(-1) | st.integers(0, 9),
+        bufsize=st.just(zlib.DEF_BUF_SIZE) | st.integers(0, zlib.DEF_BUF_SIZE),
+    )
+    def test_compress_decompress_round_trip(self, payload, level, bufsize):
+        x = zlib.compress(payload, level=level)
+        self.assertEqual(payload, zlib.decompress(x, bufsize=bufsize))
```

Comment:
@Zac-HD `zlib.decompress()` has a `kwarg : wbits` which I couldn't comprehend. The test starts breaking, if I include any wbits other than 0, 15.

Reply:
`wbits` is discussed in https://docs.python.org/3/library/zlib.html#zlib.compressobj - those specific values are 'autodetect' and the default, respectively.

I'd suggest just leaving a comment `# TODO: tests for incremental compression, wbits, strategy, zdict` (and a like to the docs as there was at the top before); you're welcome to revisit that in a future PR but let's get this chunk of work done first.